File: mashup-recruit/src/main/java/kr/mashup/branding/ui/applicant/vo/ApplicantResponse.java
package kr.mashup.branding.ui.applicant.vo;

import kr.mashup.branding.domain.applicant.Applicant;
import kr.mashup.branding.domain.applicant.ApplicantStatus;
import lombok.Data;

import java.time.LocalDate;

@Data
public class ApplicantResponse {
    private final Long applicantId;
    private final String email;
    private final String name;
    private final String phoneNumber;
    private final LocalDate birthdate;
    private final String department;
    private final String residence;
    private final ApplicantStatus status;

    public static ApplicantResponse from(Applicant applicant) {
        return new ApplicantResponse(
            applicant.getApplicantId(),
            applicant.getEmail(),
            applicant.getName(),
            applicant.getPhoneNumber(),
            applicant.getBirthdate(),
            applicant.getDepartment(),
            applicant.getResidence(),
            applicant.getStatus()
        );
    }
}

File: mashup-recruit/src/main/java/kr/mashup/branding/ui/applicant/vo/ApplicantResponse.kt
package kr.mashup.branding.ui.applicant.vo

import kr.mashup.branding.domain.applicant.Applicant
import kr.mashup.branding.domain.applicant.ApplicantStatus
import java.time.LocalDate


data class ApplicantResponse(
    val applicantId: Long,
    val email: String,
    val name: String,
    val phoneNumber: String,
    val birthdate: LocalDate,
    val department: String,
    val residence: String,
    val status: ApplicantStatus,
) {

    companion object {
        @JvmStatic
        fun from(applicant: Applicant): ApplicantResponse {
            return ApplicantResponse(
                applicant.applicantId,
                applicant.email,
                applicant.name,
                applicant.phoneNumber,
                applicant.birthdate,
                applicant.department,
                applicant.residence,
                applicant.status
            )
        }
    }
}